lib/Models/ArcGisFeatureServerCatalogItem.ts

```diff
@@ -513,73 +580,130 @@ function updateEntityWithEsriStyle(
+        const y = isDefined(symbol.yoffset) ? symbol.yoffset : 0;
+        entity.billboard.pixelOffset = new Cartesian3(x, y);
+      }
 
-    entity.point.show = new ConstantProperty(false);
-  }
+      entity.point.show = new ConstantProperty(false);
+    }
+  } else if (symbol.type === "esriSMS") {
+    // Update the styling of the Cesium Point
+    // TODO extend support for cross, diamond, square, x, triangle
+    if (entity.point && symbol.color) {
+      entity.point.color = new ConstantProperty(
+        convertEsriColorToCesiumColor(symbol.color)
+      );
+      entity.point.pixelSize = new ConstantProperty(symbol.size);
 
-  // Update the styling of the Cesium Polyline
-  if (entity.polyline && symbol.color) {
-    entity.polyline.material = new ColorMaterialProperty(
-      convertEsriColorToCesiumColor(symbol.color)
-    );
-    if (isDefined(symbol.width)) {
-      entity.polyline.width = new ConstantProperty(symbol.width);
+      if (symbol.outline) {
+        entity.point.outlineColor = new ConstantProperty(
+          convertEsriColorToCesiumColor(symbol.outline.color)
+        );
+        entity.point.outlineWidth = new ConstantProperty(symbol.outline.width);
+      }
+    }
+  } else if (symbol.type === "esriSLS") {
+    /* Update the styling of the Cesium Polyline */
+    if (entity.polyline) {
+      if (isDefined(symbol.width)) {
+        entity.polyline.width = new ConstantProperty(symbol.width);
+      }
+      const color = symbol.color ? symbol.color : defaultColor;
+      /* 
+        For line containing dashes PolylineDashMaterialProperty is used. 
+        Definition is done using the line patterns converted from hex to decimal dashPattern.
+        Source for some of the line patterns is https://www.opengl.org.ru/docs/pg/0204.html, others are created manually
+      */
+      esriPolylineStyle(entity, color, <supportedLineStyle>symbol.style);
+    }
+  } else if (symbol.type === "esriSFS") {
+    // Update the styling of the Cesium Polygon
+    if (entity.polygon) {
+      const color = symbol.color ? symbol.color : defaultColor;
+
+      // feature picking doesn't work when the polygon interior is transparent, so
+      // use an almost-transparent color instead
+      if (color[3] === 0) {
+        color[3] = 1;
+      }
+      entity.polygon.material = convertEsriColorToCesiumColor(color);
+
+      if (symbol.outline) {
+        const outlineColor = symbol.outline.color
+          ? symbol.outline.color
+          : defaultOutlineColor;
+        /* It can actually happen that entity has both polygon and polyline defined at same time,
+            check the implementation of GeoJsonCatalogItem for details. */
+        entity.polygon.outlineColor = convertEsriColorToCesiumColor(
+          outlineColor
+        );
+        entity.polygon.outlineWidth = new ConstantProperty(
+          symbol.outline.width
+        );
+        if (entity.polyline) {
+          esriPolylineStyle(entity, [], symbol.outline.style);
```

Comment:
Add comment "color value of [] is default to [255, 255, 255, 255]."

Reply:
fixed some other minor polygon issues with this one